aiohttp/test_utils.py

```diff
@@ -573,3 +573,17 @@ def mock_coro(*args, **kwargs):
         return return_value
 
     return mock.Mock(wraps=mock_coro)
+
+
+def async_data_provider(fn_data_provider):
+    """Data provider decorator, allows another callable to provide the data for the async test"""
+    def test_decorator(fn):
+        async def repl(self, *args):
+            for args in fn_data_provider():
+                try:
+                    await fn(self, *args)
+                except AssertionError:
+                    print("Assertion error caught with data set ", args)
```

Comment:
Also, if some else error happens, wouldn't it be helpful to get for which data it happened?